library/ssl_tls.c

```diff
@@ -681,20 +681,20 @@ static void ssl_calc_finished_ssl( mbedtls_ssl_context *, unsigned char *, int )
 #endif
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1) || defined(MBEDTLS_SSL_PROTO_TLS1_1)
-static void ssl_calc_verify_tls( const mbedtls_ssl_context *, unsigned char *, size_t * );
+static void ssl_calc_verify_tls( const mbedtls_ssl_context *, unsigned char [36], size_t * );
```

Comment:
Since these functions are assigned to
```
void (*calc_verify)(const mbedtls_ssl_context *, unsigned char *, size_t *);
```
from `struct mbedtls_ssl_handshake_params`, I think the prototypes here should have a pointer, not an array. Since an array type is purely indicative on a function parameter, there is no real loss in not declaring the intended buffer size.

Reply:
These `ssl_calc_verify_*` functions now use pointers in the prototypes **and** in the function definitions, because I can't mix them (pointers and arrays for the same function).
A few considerations about my latest commits:
* I kept the array parameter for the other functions because I didn't want to change the prototypes in the public headers.
* I needed to make more changes because I found other build errors switching from make to cmake.
* The only warning left is the one related to the implicit declaration of `memset`, but I saw you fixed that in #3867.